air/src/proof.rs

```diff
@@ -86,74 +86,72 @@ impl ExecutionProof {
 
 /// A set of parameters specifying how Miden VM execution proofs are to be generated.
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub struct ProofOptions {
-    pub options: WinterProofOptions,
+pub struct ProvingOptions {
     pub hash_fn: HashFunction,
+    pub exec_options: ExecutionOptions,
+    pub proof_options: WinterProofOptions,
 }
 
-impl ProofOptions {
+impl ProvingOptions {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
 
-    /// Creates a new instance of [ProofOptions] from the specified parameters.
+    /// Creates a new instance of [ProvingOptions] from the specified parameters.
     pub fn new(
-        num_queries: usize,
-        blowup_factor: usize,
-        grinding_factor: u32,
-        field_extension: FieldExtension,
-        fri_folding_factor: usize,
-        fri_max_remainder_size: usize,
+        proof_options: WinterProofOptions,
+        exec_options: ExecutionOptions,
         hash_fn: HashFunction,
     ) -> Self {
```

Comment:
I would prefer not not expose `WinterProofOptions` as a part of the public interface. So, let's keep this constructor as is, and use the builder pattern as suggested in the previous comment.